libkbfs/folder_branch_ops.go

```diff
@@ -4593,6 +4619,63 @@ func (fbo *folderBranchOps) unstageAfterFailedResolution(ctx context.Context,
 	return fbo.unstageLocked(ctx, lState)
 }
 
+func (fbo *folderBranchOps) journalBranchChange(newBID BranchID) {
+	if newBID == NullBranchID {
+		return
+	}
+
+	ctx, cancelFunc := fbo.newCtxWithFBOID()
+	defer cancelFunc()
+
+	lState := makeFBOLockState()
+	fbo.mdWriterLock.Lock(lState)
+	defer fbo.mdWriterLock.Unlock(lState)
+
+	fbo.log.CDebugf(ctx, "Journal branch change: %s", newBID)
+
+	if !fbo.isMasterBranchLocked(lState) {
+		if fbo.bid == newBID {
+			fbo.log.CDebugf(ctx, "Already on branch %s", newBID)
+			return
+		}
+		panic(fmt.Sprintf("Cannot switch to branch %s while on branch %s",
+			newBID, fbo.bid))
+	}
+
+	md, err := fbo.config.MDOps().GetUnmergedForTLF(ctx, fbo.id(), newBID)
```

Comment:
A thought: would it be easier to have the right `ImmutableRootMetadata` passed in to this delegate instead of just the branch ID? May as well, right?